java/google/registry/xjc/JaxbFragment.java

```diff
@@ -0,0 +1,82 @@
+package google.registry.xjc;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Joiner;
+import google.registry.xml.XmlException;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+
+public class JaxbFragment<T> {
+
+  /** List of packages to initialize JAXBContext. **/
+  private static final String JAXB_CONTEXT_PACKAGES = Joiner.on(":")
+      .join(Arrays.asList(
+          "google.registry.xjc.contact",
+          "google.registry.xjc.domain",
+          "google.registry.xjc.host",
+          "google.registry.xjc.mark",
+          "google.registry.xjc.rde",
+          "google.registry.xjc.rdecontact",
+          "google.registry.xjc.rdedomain",
+          "google.registry.xjc.rdeeppparams",
+          "google.registry.xjc.rdeheader",
+          "google.registry.xjc.rdeidn",
+          "google.registry.xjc.rdenndn",
+          "google.registry.xjc.rderegistrar",
+          "google.registry.xjc.smd"));
+
+  private static final Unmarshaller unmarshaller;
+  private static final Marshaller marshaller;
+
+  static {
+    try {
+      JAXBContext context = JAXBContext.newInstance(JAXB_CONTEXT_PACKAGES);
+      unmarshaller = context.createUnmarshaller();
+      marshaller = context.createMarshaller();
+    } catch (JAXBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Class<? extends Object> instanceType;
+  private byte[] frozenInstance;
+
+  public static <T> JaxbFragment<T> create(T object) throws XmlException {
+    JaxbFragment<T> fragment = new JaxbFragment<>();
+    fragment.frozenInstance = freezeInstance(object);
+    fragment.instanceType = object.getClass();
+    return fragment;
+  }
+
+  private static <T> byte[] freezeInstance(T instance) throws XmlException {
```

Comment:
Thanks. Now that I understand, I think this is an antipattern. The canonical way to approach this is to have JaxbFragment implement Serializable and implement the readObject/writeObject methods. The code will be 90% the same, but this way it's clear what the intention is rather than relying on the fact that Class<> and String are serializable types and it all happens to work.